shared/common-adapters/avatar.tsx

```diff
@@ -89,55 +84,31 @@ const followIconHelper = (size: number, followsYou: boolean, following: boolean)
   }
 }
 
-const mapStateToProps = (state: Container.TypedState, ownProps: OwnProps) => {
-  return {
+const ConnectedAvatar = Container.connect(
+  (state, ownProps: OwnProps) => ({
     _counter: state.config.avatarRefreshCounter.get(ownProps.username || ownProps.teamname || '', 0),
     _following: ownProps.showFollowingStatus ? state.config.following.has(ownProps.username || '') : false,
     _followsYou: ownProps.showFollowingStatus ? state.config.followers.has(ownProps.username || '') : false,
     _httpSrvAddress: state.config.httpSrvAddress,
     _httpSrvToken: state.config.httpSrvToken,
-  }
-}
-
-const mapDispatchToProps = (dispatch: Container.TypedDispatch, ownProps: OwnProps) => {
-  return {
-    _goToProfile: (username: string, desktopDest: 'profile' | 'tracker') =>
-      Styles.isMobile || desktopDest === 'profile'
-        ? dispatch(ProfileGen.createShowUserProfile({username}))
-        : dispatch(Tracker2Gen.createShowUser({asTracker: true, username})),
+  }),
+  (dispatch, ownProps: OwnProps) => ({
+    _goToProfile: (username: string) => dispatch(ProfileGen.createShowUserProfile({username})),
     onClick: ownProps.onEditAvatarClick ? ownProps.onEditAvatarClick : ownProps.onClick,
```

Comment:
why even do this in dispatch props?

Reply:
mergeProps evaluates username, so it should delegate it. also we likely want to avoid ownProps in mapState and mapDispatch if possible